### ISSUE TITLE
Gnome-Shell crashes in overview on X11 when using phinger-cursors

### ISSUE THREAD

**Issue by LostinSpacetime** [phisch/phinger-cursors#14]
I encounter this problem on my desktop PC (AMD RX 6800) running Ubuntu 23.04 as well as Fedora 38.
I am using the latest stable Mesa (23.1.4).
Gnome-shell crushes when in overview, and one drags any window arround. It only happens on a X11-session. With Wayland everything works just fine.

**Comment by phisch** [comment]
I'm not sure how a cursor theme would cause gnome to crash. And the cursor files provided by phinger-cursors are just in the x11 xcursor format, there isn't anything special to it.

**Comment by LostinSpacetime** [comment]
I believe it is actually a gpu-driver bug. If i turn on software cursor (Option          "SWcursor"      "on")  in xorg.conf, the gnome-shell doesn't crush. There has to be something special about phinger-cursors since there is no problem with the default (adwaita) cursor.

**Comment by phisch** [comment]
@LostinSpacetime could you by any chance upload the gnome shell error logs after a crash?

I'm not on gnome, but I know a few people that run gnome with phinger-cursors as their cursor theme and it works fine for them.

**Comment by LostinSpacetime** [comment]
Received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadCursor (invalid Cursor parameter)'.
  (Details: serial 40656 error_code 6 request_code 95 (core protocol) minor_code 0)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the MUTTER_SYNC environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the meta_x_error() function.)

**Comment by LostinSpacetime** [comment]
This happens only after also editing the file /usr/share/icons/default to "Inherits=phinger-cursors". If I don't do this, I see the Adwaita cursor on some occasions.

**Comment by phisch** [comment]
I'll leave this issue open until the next release.

Gnome shell crashing is a gnome-shell bug, but I will still try to add as many reasonable aliases as possible.

**Comment by LostinSpacetime** [comment]
I am going through the cursors/aliases Adwaita is using, and I found a few cursors that are actually missing in phinger-cursors. It's the drag and drop hand with additional symbols. (dnd-ask, dnd-copy, dnd-link, dnd-move) One can link them all to dnd-move.

The next that is missing is dotbox. I have no idea what this is for.

**Comment by LostinSpacetime** [comment]
Would it be better to file a new bug report for missing cursors/aliases?

**Comment by phisch** [comment]
@LostinSpacetime x cursor schemes can come with as many or little cursors they want, there is no defined standard, and every app might request a different name for the same cursor as long as there is an alias for it. For the new release I will restructure the cursor theme anyway, following the new wayland cursor list for the standard kit and having additional cursors on top of that, with a more exhaustive list of aliases.

Your issue is a gnome-shell bug though, it should not crash because they expect a cursor to be there that isn't. You should open a bug report there as well, though they probably won't fix it and expect you to use the adwaita cursor theme anyway.

**Comment by phisch** [comment]
@LostinSpacetime Would you mind testing if this issue still happens using the latest [2.0-beta.1 release](https://github.com/phisch/phinger-cursors/releases/tag/v2.0-beta.1)?

**Comment by phisch** [comment]
I just tested the latest GNOME release with an X11 session and the `v2.0-beta.2` release of phinger-cursors, and the issue does not seem to happen.

[gnome_x11.webm](https://github.com/phisch/phinger-cursors/assets/1282767/8e2e44e8-a75b-4719-bec4-f64685f0ebd4)

**Comment by LostinSpacetime** [comment]
2.0 has no issues so far. Now I can use it again. Thanks! :)
